packages/dashboard/src/components/reducers/main-menu-reducer.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import { StackNavigator } from 'react-components';
+import { Settings } from '../../settings';
+import { OmniPanelViewIndex } from '../dashboard';
+import { LoadingScreenProps } from '../loading-screen';
+
+type MainMenuActionFormat<T, K = void> = K extends void
+  ? {
+      type: T;
+    }
+  : {
+      type: T;
+      payload: K;
+    };
+
+export enum MainMenuActionType {
+  CurrentView = 'currentView',
+  Loading = 'loading',
+  Settings = 'settings',
+  ShowHelp = 'showHelp',
+  ShowHotkeysDialog = 'showHotkeysDialog',
+  ShowOmniPanel = 'showOmniPanel',
+  ShowSettings = 'showSettings',
+  PopView = 'popView',
+  PushView = 'pushView',
+  ResetView = 'resetView',
+  ToggleHelp = 'toggleHelp',
+  ToggleHotkeys = 'toggleHotkeys',
+  ToggleOmnipanel = 'toggleOmnipanel',
+  ToggleSettings = 'toggleSettings',
+  TourState = 'tourState',
+}
+
+export type MainMenuState = {
+  [MainMenuActionType.CurrentView]: number;
+  [MainMenuActionType.Loading]: LoadingScreenProps | null;
+  [MainMenuActionType.Settings]: Settings;
+  [MainMenuActionType.ShowHelp]: boolean;
+  [MainMenuActionType.ShowHotkeysDialog]: boolean;
+  [MainMenuActionType.ShowOmniPanel]: boolean;
+  [MainMenuActionType.ShowSettings]: boolean;
+  stackNavigator: StackNavigator<OmniPanelViewIndex>;
+  [MainMenuActionType.TourState]: boolean;
+};
+
+export type MainMenuAction =
+  | MainMenuActionFormat<'currentView', MainMenuState['currentView']>
+  | MainMenuActionFormat<'loading', MainMenuState['loading']>
+  | MainMenuActionFormat<'settings', MainMenuState['settings']>
+  | MainMenuActionFormat<'showHelp', MainMenuState['showHelp']>
+  | MainMenuActionFormat<'showHotkeysDialog', MainMenuState['showHotkeysDialog']>
+  | MainMenuActionFormat<'showOmniPanel', MainMenuState['showOmniPanel']>
+  | MainMenuActionFormat<'showSettings', MainMenuState['showSettings']>
+  | MainMenuActionFormat<'popView'>
+  | MainMenuActionFormat<'pushView', MainMenuState['currentView']>
+  | MainMenuActionFormat<'resetView'>
+  | MainMenuActionFormat<'toggleHelp'>
+  | Partial<MainMenuActionFormat<'toggleHotkeys'>>
+  | Partial<MainMenuActionFormat<'toggleOmnipanel'>>
+  | Partial<MainMenuActionFormat<'toggleSettings'>>
```

Comment:
I don't think we need the `Partial` here as there is no payload in these actions

Reply:
It was a leftOver thanks!